src/ir/stack-utils.cpp

```diff
@@ -160,6 +160,35 @@ StackSignature StackSignature::operator+(const StackSignature& next) const {
   return sig;
 }
 
+StackSignature StackSignature::getLeastUpperBound(StackSignature a,
+                                                  StackSignature b) {
+  // Assert an approximation of the conditions for having a LUB
+  assert(a.unreachable || b.unreachable ||
+         (a.params.size() == b.params.size() &&
+          a.results.size() == b.results.size()));
```

Comment:
In what sense is an LUB an LUB if the number of params or results are shorter? Do we fill the missing slots in the unreachable one as needed, so it can be shorter (and it must be the shorter one, if so - we could assert on that)?

Reply:
Exactly, missing items can be filled in only on unreachable stack signatures. I'll add an assertion as you suggest.